### ISSUE TITLE
Allow callback function for Key

### ISSUE THREAD

**Issue by andyinabox** [appleboy/gin-jwt#234]
**TL;DR:** Add a `KeyFunc` callback that takes the JWT as an argument and can be used in place of `Key`

I'm working on an implementation that uses a JSON Web Key (JWK) to validate my JWTs. There's a pretty good breakdown of this flow in ["Navigating RS256 and JWKS"](https://auth0.com/blog/navigating-rs256-and-jwks/) (auth0 blog).

Since my implementation will use Gin, it would be great to use gin-jwt, but I need to be able to fetch the correct JWK using the `kid` included in the JWT, which means I'd need to fetch it at execution time.

Auth0 has a module [`jwtmiddleware`](https://github.com/auth0/go-jwt-middleware) that uses a callback `ValidationKeyGetter` to get the key, rather than just using a string. It would be great to be able to use gin-jwk in the same way.

I think it could be as simple as having a `KeyFunc` callback, and if the `Key` value is `nil` then it will try to use the callback function. Usage would look something like

```go
// using Token from the jwk-go module
KeyFunc: func(token *jwt.Token) []byte {
  var key []byte

 // fetch the JWK, retrieve the key from it

  return key
},
```

**Comment by lihaotian0607** [comment]
I think it ’s good, but why was it rejected?
